variables.tf

```diff
@@ -57,6 +57,12 @@ variable "lifecycle_rule" {
   description = "List of maps containing lifecycle management configuration settings"
 }
 
+variable "logging" {
+  description = "Map containing access bucket logging configuration."
+  type        = map(string)
+  default     = {}
```

Comment:
Please use ab object instead of a map. And please also fix the ordering 😏

Reply:
just tried to switch it out to an object, but got stuck. how do I make 

`dynamic logging {
    for_each = local.logging_configuration

    content {
      target_bucket = local.logging_configuration.target_bucket
      target_prefix = local.logging_configuration.target_prefix
    }
  }`

the target prefix optional when using an object? seems like I must declare both or neither. whereas the target prefix is an entirely optional thing we might not want to use.